api/grpc/service.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package pb;
+
+
+service WorkerService {
+  rpc Start        (JobRequest         ) returns (JobStatus       ) {}
+  rpc Status       (StatusRequest      ) returns (JobStatus       ) {}
+  rpc Stop         (StopRequest        ) returns (JobStatus       ) {}
+  rpc GetOuput     (GetOutputRequest   ) returns (stream JobOutput) {}
+}
+
+message StartRequest {
+  string id = 1;
+}
+
+message StatusRequest {
+  string id = 1;
+}
+
+message StopRequest {
+  string id = 1;
+}
+
+message GetOutputRequest {
+  string id = 1;
+}
+
+message JobOutput {
+  string output = 1;
+  OutputChannel channel = 2;
+}
+
+message JobRequest {
+  string cmd           = 1;
+  repeated string args = 2;
+}
+
+message JobStatus {
+  string id              = 1; // the id of the job
+  int64 PID              = 2; // the process id of the process running in the system
+  string cmd             = 3; // a single string containing the full command and arguments passed in the job request
+  JobState state         = 4; // the current state of the process
+  int64 exitCode         = 5; // the exit code of the process
+  string error           = 6; // the error if any, when the process reached a final state
+  int64 startTime        = 7;
+  int64 stopTime         = 8;
+}
+
+enum OutputChannel {
+  StdOut = 0;
+  StdErr = 1;
+}
+
+enum JobState {
+  Starting    = 0; // process is starting
```

Comment:
What's this state for? For simplicity, the server can just start the job as soon as it receives the request.

Reply:
Thanks. I will keep it as internal state for the process